sktree/tests/test_honest_forest.py

```diff
@@ -263,6 +263,27 @@ def test_impute_posteriors(honest_prior, val):
         ), f"Failed with {honest_prior}, prior {clf.estimators_[0].empirical_prior_}"
 
 
+@pytest.mark.parametrize(
+    "honest_prior, val",
+    [
+        ("ignore", np.nan),
+    ],
+)
+def test_ignore_posteriors(honest_prior, val):
+    X = rng.normal(0, 1, (100, 2))
+    y = [0] * 75 + [1] * 25
+    clf = HonestForestClassifier(
+        honest_fraction=0.5, random_state=0, honest_prior=honest_prior, n_estimators=100
+    )
+    clf = clf.fit(X, y)
+
+    y_proba = clf.predict_proba(X)
+
+    assert (
+        len(np.where(np.isnan(y_proba[:, 0]))[0]) < 10
+    ), f"Failed with {honest_prior}, prior {clf.estimators_[0].empirical_prior_}"
+
```

Comment:
Perhaps it should just assert that none of `y_pred` are nans? I think with enough trees this is possible. 

It would be nice to add an in-line comment specifying this is a requirement for the test to pass too.